### DISCUSSION TITLE
Revert to commit and delete history Boîte de réception 	x

### DISCUSSION THREAD

**Discussion by sigiaccio** [community/community#21782] [New to GitHub]
<p>Hello,<br>
I made a commit ‘merge and squash’ but it didn’t work (certainly an error when I clicked)</p>
<p>Can you tell me how to delete the history?<br>
Not all the history, just the history of the last commit.</p>
<p>I have done this :</p>
<ol>
<li>revert in web to xxxxxxx commit<br>
2 .Git push origin +xxxxxxx^:master</li>
<li>git reset HEAD^ --hard</li>
<li>git push origin -f</li>
</ol>
<p>I would like do this</p>
<ol>
<li>revert to this commit xxxxxxx with clear history </li>
<li>delete the merge to master from branch dvplp</li>
<li>delete 204 commit from branche dvplp</li>
</ol>
<p>Thanks</p>

**Comment by sigiaccio** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c913543d6b7f7547030493cee1b954dd1e45cab4.png" data-download-href="/uploads/short-url/sGNnNEy0lWulAhUYzlVGW6Bsv08.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/42961987/181066855-60a68f6c-1bf6-4370-92d4-60de5533de35.png" alt="image.png" data-base62-sha1="sGNnNEy0lWulAhUYzlVGW6Bsv08" width="690" height="467" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/c913543d6b7f7547030493cee1b954dd1e45cab4_2_690x467.png, https://aws1.discourse-cdn.com/github/original/2X/c/c913543d6b7f7547030493cee1b954dd1e45cab4.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/c/c913543d6b7f7547030493cee1b954dd1e45cab4.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c913543d6b7f7547030493cee1b954dd1e45cab4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">773×524 34.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Wabri** [comment, marked as answer]
<p>Ok, this is a little danger but i will try with an example.</p>
<p><strong>1.</strong> This is the history of my master:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/26a79e795149bd2e13a90895bf9b7e58b2398801.png" data-download-href="/uploads/short-url/5vXguxTVUHqIbMyZRh75t1Rbw7T.png?dl=1" title="reset1.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181066856-6d47a271-64fb-4f51-90ad-9893113c8f13.png" alt="reset1.png" data-base62-sha1="5vXguxTVUHqIbMyZRh75t1Rbw7T" width="690" height="146" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/26a79e795149bd2e13a90895bf9b7e58b2398801_2_690x146.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/26a79e795149bd2e13a90895bf9b7e58b2398801_2_1035x219.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/26a79e795149bd2e13a90895bf9b7e58b2398801_2_1380x292.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/26a79e795149bd2e13a90895bf9b7e58b2398801_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">reset1.png</span><span class="informations">1441×306 29 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><strong>2.</strong> I want to delete the second commit with description <strong><em>“this commit need to be deleted”</em></strong>. The first thing to do is to get the <strong>commit ID</strong> of the previous commit, in my case the commit is <strong><em>“reset here”</em></strong>. You can get the commit id by the web UI:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/e79ed4a732e79dcb5992921715e3ea796ce5a3a9.png" data-download-href="/uploads/short-url/x30Avsf7psrJ3CP9CMCYDagONoB.png?dl=1" title="reset2.1.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181066857-91834879-e790-4509-adfe-dac38726fb11.png" alt="reset2.1.png" data-base62-sha1="x30Avsf7psrJ3CP9CMCYDagONoB" width="690" height="155" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/e79ed4a732e79dcb5992921715e3ea796ce5a3a9_2_690x155.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/e79ed4a732e79dcb5992921715e3ea796ce5a3a9_2_1035x232.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/e/e79ed4a732e79dcb5992921715e3ea796ce5a3a9_2_1380x310.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e79ed4a732e79dcb5992921715e3ea796ce5a3a9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">reset2.1.png</span><span class="informations">1508×339 64 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/b548e564788a220be95e223a69e7c0cb04149bfc.png" data-download-href="/uploads/short-url/pRIBhcpZMl1BrOsGyCNDT9MVPwo.png?dl=1" title="reset2.2.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181066860-2328864f-875f-4f92-b9b3-48ad9706f1d3.png" alt="reset2.2.png" data-base62-sha1="pRIBhcpZMl1BrOsGyCNDT9MVPwo" width="690" height="146" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/b548e564788a220be95e223a69e7c0cb04149bfc_2_690x146.png, https://aws1.discourse-cdn.com/github/optimized/2X/b/b548e564788a220be95e223a69e7c0cb04149bfc_2_1035x219.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/b/b548e564788a220be95e223a69e7c0cb04149bfc_2_1380x292.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/b548e564788a220be95e223a69e7c0cb04149bfc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">reset2.2.png</span><span class="informations">1441×306 58.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Or with command line:</p>
<pre><code>git log
</code></pre>
<p>with output:</p>
<pre><code>commit 2e45e59114bbb7f6b36b8e8598415a96690190e4
Author: Gabriele Puliti &lt;gabriele.puliti@gmail.com&gt;
Date: Wed Jan 30 16:57:56 2019 +0100

    i want this file on history

commit a18476877494ec36e30fa4edc0850bbf36f08100
Author: Gabriele Puliti &lt;gabriele.puliti@gmail.com&gt;
Date: Wed Jan 30 16:57:21 2019 +0100

    this commit need to be delete

commit_&lt;u&gt;&lt;strong&gt;dd9723792537280460bfa67d8319086fed79e1f&lt;/strong&gt;&lt;/u&gt;_&lt;u&gt;&lt;strong&gt;&amp;lt;------------------ THIS IS THE ID&lt;/strong&gt;&lt;/u&gt;Author: Gabriele Puliti &lt;gabriele.puliti@gmail.com&gt;
Date: Wed Jan 30 16:56:29 2019 +0100

    reset here
</code></pre>
<p>Copy this value.</p>
<p><strong>3.</strong> And now <strong>reset to this commit by paste that ID</strong> after the reset command:</p>
<pre><code>git reset dd9723792537280460bfa67d8319086fed79e1fa
</code></pre>
<p>this move the local history to this commit.</p>
<p><strong>4.</strong> If you dont want the <strong>changes made by the last commit</strong> you can jump to the next step (point 5). Otherwise you need to create a new branch, for example called “helper”:</p>
<pre><code>git checkout -B helper
</code></pre>
<p>and cherry-pick the last commit, in my case the commit with description <strong><em>“i want this file on history”</em></strong> , using the commit ID of this commit:</p>
<pre><code>git cherry-pick 2e45e59114bbb7f6b36b8e8598415a96690190e4
</code></pre>
<p><em>(Remeber that the working tree need to be empty, so commit or remove the dirty files.)</em></p>
<p>Now you need to merge this new branch into master, so checkout to master:</p>
<pre><code>git checkout master
</code></pre>
<p>And merge:</p>
<pre><code>git merge helper
</code></pre>
<p>You can now remove the “helper” branch:</p>
<pre><code>git branch -D helper
</code></pre>
<p><strong>5.</strong> Last step you need to force push to remote:</p>
<pre><code>git push --force origin master
</code></pre>
<p>This is the last command you need to run.</p>
<p><strong>6.</strong> If you follow the point 4 you have:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/e57a5616ee81735a967481ca0c201ea506b2ad45.png" data-download-href="/uploads/short-url/wK3rAEG1MuEBh6ZoiQZJJ0dACSF.png?dl=1" title="reset3.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181066862-2b4d41cc-bebf-495b-9e4e-7e7663ece5d8.png" alt="reset3.png" data-base62-sha1="wK3rAEG1MuEBh6ZoiQZJJ0dACSF" width="690" height="100" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/e57a5616ee81735a967481ca0c201ea506b2ad45_2_690x100.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/e57a5616ee81735a967481ca0c201ea506b2ad45_2_1035x150.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/e/e57a5616ee81735a967481ca0c201ea506b2ad45_2_1380x200.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e57a5616ee81735a967481ca0c201ea506b2ad45_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">reset3.png</span><span class="informations">1441×209 19.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Otherwise you only have:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/098d3343de4ca0f827bc6d0f1527e7f66ed53e91.png" data-download-href="/uploads/short-url/1muOG1r01jY6UbpcXMBFnX7ivyp.png?dl=1" title="reset4.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/12409541/181066863-d9ceda73-9d96-407b-84d3-2ec5e4a3ba4b.png" alt="reset4.png" data-base62-sha1="1muOG1r01jY6UbpcXMBFnX7ivyp" width="690" height="59" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/098d3343de4ca0f827bc6d0f1527e7f66ed53e91_2_690x59.png, https://aws1.discourse-cdn.com/github/optimized/2X/0/098d3343de4ca0f827bc6d0f1527e7f66ed53e91_2_1035x88.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/0/098d3343de4ca0f827bc6d0f1527e7f66ed53e91_2_1380x118.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/098d3343de4ca0f827bc6d0f1527e7f66ed53e91_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">reset4.png</span><span class="informations">1449×125 9.61 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Hope this helps :slightly_smiling_face:  </p>
<p>-Gabriele-</p>
<blockquote>
<p><strong>Mark helpful posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great content!</strong></p>
</blockquote>

**Comment by sigiaccio** [comment]
<p>It’s perfect - it’s working. Thanks for all documentation.</p>
<p>I found also this command :</p>
<p>git rebase --onto master~1 master </p>
<p>git push origin master -f</p>
<p>Source : <a href="https://blog.hadrien.eu/2014/05/07/supprimer-un-commit-dans-lhistorique-de-git/" rel="nofollow noopener">https://blog.hadrien.eu/2014/05/07/supprimer-un-commit-dans-lhistorique-de-git/</a></p>
<p>Thanks for support!</p>
<p>Have a nice day</p>

**Comment by stirpude** [comment]
<p>Hi Wabri,</p>
<p>This was really very helpful, thank you.</p>
<p>Regards,</p>
<p>Shruti</p>
